.tools/validation/validate_doc_metadata.py

```diff
@@ -161,53 +170,63 @@ def validate_files(schema_name, meta_names, validators):
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument(
-        "doc_gen", help="The folder that contains schema and metadata files.")
+        "--doc_gen",
+        default=f"{Path(__file__).parent / '..' / '..' / '.doc_gen'}",
+        help="The folder that contains schema and metadata files.",
+        required=False,
+    )
     args = parser.parse_args()
 
-    with open(os.path.join(args.doc_gen, 'metadata/sdks.yaml')) as sdks_file:
-        sdks_yaml = yaml.safe_load(sdks_file)
+    # with open(os.path.join(args.doc_gen, "metadata/sdks.yaml")) as sdks_file:
+    #     sdks_yaml: dict[str, any] = yaml.safe_load(sdks_file)
 
-    with open(os.path.join(args.doc_gen, 'metadata/services.yaml')) as services_file:
+    with open(os.path.join(args.doc_gen, "metadata/services.yaml")) as services_file:
         services_yaml = yaml.safe_load(services_file)
 
-    with open(os.path.join(args.doc_gen, 'metadata/curated/sources.yaml')) as curated_sources_file:
+    with open(
+        os.path.join(args.doc_gen, "metadata/curated/sources.yaml")
+    ) as curated_sources_file:
         curated_sources_yaml = yaml.safe_load(curated_sources_file)
 
     validators = DefaultValidators.copy()
     ServiceName.services = services_yaml
     SourceKey.curated_sources = curated_sources_yaml
     ExampleId.services = services_yaml
-    BlockContent.block_names = os.listdir(os.path.join(args.doc_gen, 'cross-content'))
+    BlockContent.block_names = os.listdir(os.path.join(args.doc_gen, "cross-content"))
     validators[ServiceName.tag] = ServiceName
     validators[ServiceVersion.tag] = ServiceVersion
     validators[SourceKey.tag] = SourceKey
     validators[ExampleId.tag] = ExampleId
     validators[BlockContent.tag] = BlockContent
     validators[String.tag] = StringExtension
 
+    schema_root = Path(__file__).parent / "schema"
+
     # Validate sdks.yaml file.
-    schema_name = os.path.join(args.doc_gen, 'validation/sdks_schema.yaml')
-    meta_names = glob.glob(os.path.join(args.doc_gen, 'metadata/sdks.yaml'))
+    schema_name = schema_root / "sdks_schema.yaml"
```

Comment:
What's the `/` syntax here?

Reply:
[pathlib.Path](https://docs.python.org/3/library/pathlib.html) file paths.